test/Microsoft.AspNetCore.Authentication.Test/Google/GoogleMiddlewareTests.cs

```diff
@@ -781,6 +789,13 @@ private static TestServer CreateServer(GoogleOptions options, Func<HttpContext,
                         {
                             await context.Authentication.ChallengeAsync("Google");
                         }
+                        else if (req.Path == new PathString("/tokens"))
+                        {
+                            var authContext = new AuthenticateContext(TestExtensions.CookieAuthenticationScheme);
```

Comment:
Well, I'm totally fine using this pattern with protected members you override (it's not really different from calling a method on event handler args) but I think it's a terrible pattern for public APIs, as it's counterintuitive: instantiating a context object and expect the method you call to alter it is rather unusual. 

> Le 17 févr. 2016 à 21:55, Hao Kung <notifications@github.com> a écrit :
> 
> In test/Microsoft.AspNetCore.Authentication.Test/Google/GoogleMiddlewareTests.cs:
> 
> > @@ -781,6 +789,13 @@ private static TestServer CreateServer(GoogleOptions options, Func<HttpContext,
> >                          {
> >                              await context.Authentication.ChallengeAsync("Google");
> >                          }
> > +                        else if (req.Path == new PathString("/tokens"))
> > +                        {
> > +                            var authContext = new AuthenticateContext(TestExtensions.CookieAuthenticationScheme);
> Internally we have a fairly widespread pattern of creating a context, and passing it to a method, I don't think its THAT bad. But its not the best public API no :)
> 
> —
> Reply to this email directly or view it on GitHub.
>